### ISSUE TITLE
Desktop: System.Linq.Parallel.Tests.OrderByThenByTests.OrderByDescending_ExtremeComparer failed

### ISSUE THREAD

**Issue by Jiayili1** [dotnet/corefx#18140]
Failed test: System.Linq.Parallel.Tests.OrderByThenByTests.OrderByDescending_ExtremeComparer

Detail: https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netfx_windows_nt_debug/13/testReport/System.Linq.Parallel.Tests/OrderByThenByTests/OrderByDescending_ExtremeComparer_labeled__Enumerable_Range_Ordered__count__2_/

~~~
MESSAGE:
Assert.InRange() Failure\r\nRange: (0 - 0)\r\nActual: 1
+++++++++++++++++++
STACK TRACE:
at System.Linq.Parallel.Tests.OrderByThenByTests.OrderByDescending_ExtremeComparer(Labeled`1 labeled, Int32 count) in D:\j\workspace\outerloop_net---903ddde6\src\System.Linq.Parallel\tests\QueryOperators\OrderByThenByTests.cs:line 384
~~~

**Comment by danmosemsft** [comment]
@alexperovich can you have a quick take on whether this is a concern for 2.0?

**Comment by alexperovich** [comment]
This is the same underlying issue and fix as #18141
